exec-analysis/src/main/java/com/capitalone/dashboard/exec/collector/PortfolioCollector.java

```diff
@@ -359,4 +400,45 @@ public void run() {
     public void onStartup() {
         taskScheduler.schedule(this, new CronTrigger(setting.getCron()));
     }
+
+    public String getPortfolioWithThumbnails(List<Portfolio> portfolioList) {
+        ldapHandler.initLdaptemplate();
+        setLdapTemplate(LdapHandler.getLdaptemplate());
+        PortfolioThumbnail portfolioThumbnail = null;
+        List<PortfolioThumbnail> portfolioListThumbnail = new ArrayList<>();
+        for(Portfolio p :  portfolioList) {
+            portfolioThumbnail = new PortfolioThumbnail();
+            String pName = p.getName();
+            String lob = p.getLob();
+            String photoString = null;
+            String lastName = pName.substring(pName.indexOf(" ") + 1);
+            String firstName = pName.substring(0, pName.indexOf(" "));
+            String fullName = lastName + ", " + firstName;
+            if (!fullName.isEmpty()) {
```

Comment:
This is deeply nested. Consider the fail fast approach to remove layers of nesting

Reply:
Modified code and avoid nested if by calling method.